jme3-core/src/com/jme3/gde/core/assets/actions/MergeAnimationsAction.java

```diff
@@ -0,0 +1,208 @@
+    
+    private static final String CANCEL = "Cancel";
+    private final List<SpatialAssetDataObject> spatials;
+    private final Logger logger;
+
+    public MergeAnimationsAction(List<SpatialAssetDataObject> context) {
+        this.spatials = context;
+        logger = Logger.getLogger(MergeAnimationsAction.class.getName());
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        if (spatials.size() == 1) {
+            DialogDisplayer.getDefault().notify(new NotifyDescriptor.Message(
+                            "Must select more than one spatial",
+                            NotifyDescriptor.ERROR_MESSAGE));
+            return;
+        }
+        
+        final Object selectedSpatial = createSelector().getValue();
+        
+        if (selectedSpatial == null || selectedSpatial.equals(CANCEL)) {
+            logger.log(Logger.Level.INFO, "Operation cancelled by user.");
+            return;
+        }
+        
+        final SpatialAssetDataObject targetAsset = findSpatial(selectedSpatial.toString());
+        
+        if (targetAsset == null) {
+            logger.log(Logger.Level.INFO, "Operation failed. No spatial.");
+            return;
+        }
+        
+        final Spatial targetSpatial = targetAsset.loadAsset();
+        final AnimComposer targetAnimComposer = findAnimComposer(targetSpatial, null);
+        
+        if (targetAnimComposer == null) {
+            DialogDisplayer.getDefault().notify(new NotifyDescriptor.Message(
+                            String.format("%s has no AnimComposer.", targetSpatial),
+                            NotifyDescriptor.ERROR_MESSAGE));
+            return;
+        }
+        final Spatial targetAnimComposerSpatial = targetAnimComposer.getSpatial();
+        for (final Iterator<SpatialAssetDataObject> it = spatials.iterator(); it.hasNext();) {
+            final SpatialAssetDataObject spatialAssetDataObject = it.next();
+            if(spatialAssetDataObject.getName().equals(selectedSpatial)) {
+                continue;
+            }
+            Spatial sourceSpatial = spatialAssetDataObject.loadAsset();
+            final AnimComposer sourceAnimComposer = findAnimComposer(sourceSpatial, null);
+            if (sourceAnimComposer == null) {
+                DialogDisplayer.getDefault().notify(new NotifyDescriptor.Message(
+                            String.format("%s has no AnimComposer.", sourceSpatial),
+                            NotifyDescriptor.ERROR_MESSAGE));
+                return;
+            }
+            copyClips(sourceAnimComposer, targetAnimComposer, targetAnimComposerSpatial.getControl(SkinningControl.class).getArmature());
+        }
+        logger.log(Logger.Level.INFO, "Merging animations done. Saving.");
+        try {
+            targetAsset.saveAsset();
+        } catch (IOException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+    }
+    
+    private static AnimClip retargetClip(
```

Comment:
The condition `MyAnimation.isJointTrack(animTrack)` is too restrictive and does not consider the case where the target of the `AnimTracks` is a `Node` and not a `Joint` (see animation files downloaded from Mixamo 'without skeleton', which are much more efficient because they take up less disk space). Please modify the code in this way to handle both cases:

 ```java
private static AnimClip retargetClip(AnimClip sourceClip, Armature targetArmature, String clipName) {

	// Create a list to hold the new tracks
	SafeArrayList<AnimTrack> tracks = new SafeArrayList<>(AnimTrack.class);

	// Iterate through each track in the source clip
	for (AnimTrack animTrack : sourceClip.getTracks()) {

		TransformTrack sourceTrack = (TransformTrack) animTrack;
		String targetName = getTargetName(sourceTrack.getTarget());
		Joint target = targetArmature.getJoint(targetName);

		if (target != null) {
			// Clone the source track and set the new target joint
			TransformTrack newTrack = sourceTrack.jmeClone();
			newTrack.setTarget(target);
			tracks.add(newTrack);

		} else {
			logger.log(Level.WARNING, "Joint not found in the target Armature: {0}", targetName);
		}
	}

	// Create a new animation clip with the specified name and set its tracks
	AnimClip newClip = new AnimClip(clipName);
	newClip.setTracks(tracks.getArray());

	logger.log(Level.INFO, "Created new AnimClip {0} with {1} tracks out of {2} from the source clip",
			new Object[]{clipName, tracks.size(), sourceClip.getTracks().length});
	return newClip;
}

private static String getTargetName(HasLocalTransform target) {
	String targetName = null;
	// Determine the name of the target based on its type
	if (target instanceof Node) {
		targetName = ((Node) target).getName();
	} else if (target instanceof Joint) {
		targetName = ((Joint) target).getName();
	}

	return targetName;
}
```

Reply:
Note: This change optimizes the current solution and is independent of the Wes library.

Reply:
Thanks. I've merged these changes and will try them out later. Next time maybe you can create a PR? ;)